astropy/stats/sigma_clipping.py

```diff
@@ -402,7 +437,13 @@ def _sigmaclip_fast(
         else:
             return result
 
-    def _sigmaclip_noaxis(self, data, masked=True, return_bounds=False, copy=True):
+    def _sigmaclip_noaxis(
+        self,
+        data: ArrayLike,
+        masked: bool | None = True,
+        return_bounds: bool | None = False,
+        copy: bool | None = True,
+    ) -> ArrayLike | tuple[ArrayLike, float, float]:
```

Comment:
It's not actually ArrayLike, it's NDArray | MaskedArray.

Reply:
Fixed in [70b2007](https://github.com/astropy/astropy/pull/16562/commits/70b2007960c680ca2b474f8274f008b1f2030165)